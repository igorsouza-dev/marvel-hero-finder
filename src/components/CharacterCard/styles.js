import styled from 'styled-components';
import { screenSize } from 'styles/variables';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: center;
  width: ${(props) => props.size}px;
  height: 300px;

  background-image: url(${(props) => props.url});
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  border-radius: 4px;
  box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.5);
  border: solid 2px #363131;
  transition: box-shadow 0.5s ease-out;
  flex-shrink: 0;
  @media only screen and (max-width: ${screenSize.mobileL}) {
    margin: 5px;
  }
  &:hover {
    cursor: pointer;
    box-shadow: 0px 0px 20px #b80000;
  }
  img {
    border-radius: 4px;
    height: 100%;
  }
  div {
    background: #b80000;
    width: 100%;
    display: flex;
    height: 48px;
    align-items: center;
    justify-content: center;
    overflow: hidden;
    strong {
      color: #fff;
      text-transform: uppercase;
      text-overflow: clip;
      text-align: center;
    }
    border-radius: 0 0 4px 4px;
  }
`;
