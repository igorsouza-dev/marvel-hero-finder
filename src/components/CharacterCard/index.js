import React from 'react';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';

import { Container } from './styles';

function CharacterCard({ character, size }) {
  const history = useHistory();
  return (
    <Container
      size={size}
      url={`${character.thumbnail.path}/portrait_uncanny.${character.thumbnail.extension}`}
      onClick={() => history.push(`/details/${character.id}`)}
    >
      <div>
        <strong>{character.name}</strong>
      </div>
    </Container>
  );
}

CharacterCard.propTypes = {
  character: PropTypes.shape({
    thumbnail: PropTypes.shape({
      path: PropTypes.string.isRequired,
      extension: PropTypes.string.isRequired,
    }).isRequired,
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
  size: PropTypes.number,
};

CharacterCard.defaultProps = {
  size: 200,
};

export default CharacterCard;
