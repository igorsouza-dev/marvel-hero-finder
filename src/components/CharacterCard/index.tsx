import React from 'react';
import { useHistory } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';

import Character from 'types/Character';
import { setCharacter } from 'store/actions';
import { Container } from './styles';

interface PropsCharacterCard {
  character: Character;
}

const CharacterCard: React.FC<PropsCharacterCard> = ({ character }) => {
  const dispatch = useDispatch();
  const history = useHistory();

  function showDetails(selectedCharacter: Character) {
    dispatch(setCharacter(selectedCharacter));
    history.push(`/details/${selectedCharacter.id}`);
  }

  return (
    <Container url={character.thumbnail} onClick={() => showDetails(character)}>
      <div>
        <strong>{character.name}</strong>
      </div>
    </Container>
  );
};

CharacterCard.propTypes = {
  character: PropTypes.shape({
    thumbnail: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
  }).isRequired,
};

export default CharacterCard;
