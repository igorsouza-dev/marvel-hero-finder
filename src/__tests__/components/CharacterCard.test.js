import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterCard from 'components/CharacterCard';

describe('CharacterCard', () => {
  afterEach(cleanup);
  test('renders a hero card', () => {
    const character = {
      id: 123456,
      name: 'Iron Man',
      thumbnail: {
        path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
        extension: 'jpg',
      },
    };
    const { container, getByText } = render(<CharacterCard character={character} />);
    const linkElement = getByText(/Iron Man/i);
    expect(linkElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
