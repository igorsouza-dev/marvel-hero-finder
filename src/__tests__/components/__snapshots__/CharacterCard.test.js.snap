// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard renders a hero card 1`] = `
<div
  class="sc-bdVaJa dKPgmu"
>
  <div>
    <strong>
      Iron Man
    </strong>
  </div>
</div>
`;
